transformations/mixed_language_perturbation/transformation.py

```diff
@@ -0,0 +1,89 @@
+import random
+import string
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+"""
+Mixed Language Perturbation
+This perturbation translates words in the text from English to other languages (e.g., German). It can be used to test the robustness of a model in a multilingual setting.
+"""
+
+def translate(model, tokenizer, text, src_lang, target_lang):
+    tokenizer.src_lang = src_lang
+    encoded = tokenizer(text, return_tensors="pt")
+    generated_tokens = model.generate(**encoded, forced_bos_token_id=tokenizer.get_lang_id(target_lang))
+    res = tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+    return res
+
+def mixed_language(model, tokenizer, text, prob_mix=0.1, src_lang="en", trg_lang="fr", seed=0):
+    random.seed(seed)
+
+    words = text.split()
+    mixed_text = ""
+    for word in words:
+        if mixed_text != "":
+            mixed_text += " "
+        if random.random() < prob_mix:
+            plain_word = word.translate(str.maketrans('', '', string.punctuation)).strip()
+
+            if plain_word == "":
+                continue
+
+            mixed_text += translate(model, tokenizer, plain_word, src_lang, trg_lang)[0]
+
+            if word[-1] in string.punctuation:
+                mixed_text += word[-1]
+        else:
+            mixed_text += word
+
+    print(text, mixed_text)
+    return mixed_text
+
+class MixedLanguagePerturbation(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["en"]
+    tgt_languages = ["es", "de", "fr", "zh"]
+
```

Comment:
It would be good to add a `heavy=True` mark just like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/0672d7163338c3c744c9964e28084372781ab349/transformations/back_translation/transformation.py#L10). Therefore, when running pytest, it would by default ignore the heavy transformations so that the pretrained model won't be loaded.

Reply:
Thank you 👍 We added `heavy=true` in our latest commit.